openscap_report/report_generators/html_templates/js/oval_graph_generation_script.js

```diff
@@ -532,6 +530,57 @@ function get_OVAL_test_info(test_info) {
     }
     table.appendChild(get_table_header(objects));
     table.appendChild(get_table_body(objects));
+}
+
+function get_OVAL_state_labels(oval_state) {
+    const div = DIV.cloneNode();
+    const text = P.cloneNode();
+    text.textContent ='This is expected state of OVAL object: ';
+    text.appendChild(BR.cloneNode());
+
+    div.appendChild(text);
+    div.appendChild(get_label("", oval_state.state_id));
+    div.appendChild(get_label("pf-m-blue", oval_state.comment));
+    return div;
+}
+
+function generate_OVAL_state(test_info, div) {
+    if (test_info.oval_state === null) {
+        return;
+    }
+    div.appendChild(BR.cloneNode());
+    div.appendChild(get_OVAL_state_labels(test_info.oval_state));
+    const table_div = DIV.cloneNode();
+    table_div.className = "pf-c-scroll-inner-wrapper oval-test-detail-table";
+    div.appendChild(table_div);
+    const table = TABLE.cloneNode();
+    table.className = "pf-c-table pf-m-compact pf-m-grid-md";
+    table.setAttribute("role", "grid");
+    table_div.appendChild(table);
+
+    const objects = [];
+    for (const data of test_info.oval_state.state_data) {
+        objects.push(filter_object(data, test_info.oval_state));
+    }
+    table.appendChild(get_table_header(objects));
+    table.appendChild(get_table_body(objects));
+}
+
+
+function get_OVAL_test_info(test_info) {
+    const div = DIV.cloneNode();
+    div.className = "pf-c-accordion__expanded-content-body";
+    div.appendChild(get_label("", test_info.test_id));
+    div.appendChild(get_label("pf-m-blue", test_info.comment));
+    if (test_info.check) {
+        div.appendChild(get_label("pf-m-cyan", `Check atribute: ${test_info.check}`));
```

Comment:
It's nice that it shows now the value of the `check` and `check_existence` attributes. But I'm afraid that no user except from the content authors know what this means. It would be better to show also the meaning of that attribute.

It can, for example `check_existence="none_exist"`, say: This check requires that no such object can exist on the system.

You can find the possible values and their meaning in the OVAL specification. The report generate could translat attributus values to some helpful text.